source/m2mconnectionsecuritypimpl.cpp

```diff
@@ -146,22 +152,23 @@ int M2MConnectionSecurityPimpl::start_handshake(){
         return M2MConnectionHandler::CONNECTION_ERROR_WANTS_READ;
     }
 
-    if( ret != PAL_SUCCESS ) { //We loose the original error here!
-        ret = -1;
-    }else {
-        if( ( /*_flags*/PAL_SUCCESS != pal_sslGetVerifyResult( _ssl ) ) ) { //TODO: check if flags used somewhere?
-            ret = -1;
-        }
+    if(ret != PAL_SUCCESS){ //We loose the original error here!
+        return -1;
+    }
+
+    if(PAL_SUCCESS != pal_sslGetVerifyResult( _ssl )){
+        return -1;
     }
-    tr_debug("M2MConnectionSecurityPimpl::start_handshake - OUT");
+
     return ret;
 }
 
-int M2MConnectionSecurityPimpl::connect(M2MConnectionHandler* connHandler){
-
+int M2MConnectionSecurityPimpl::connect(M2MConnectionHandler* connHandler)
+{
     tr_debug("M2MConnectionSecurityPimpl::connect");
-    int ret=-1;
-    if(!_init_done){
+    int ret = -1;
+
+    if(2 != _init_done){
```

Comment:
Where is this hard-coded 2 coming from? Needs to be literal or in minimum why its 2? 
it's a much stricter comparision than the previous !_init_done, why?